### PR TITLE
Issue #11: Created SQlite DB for the ShuttleTimes of both campuses

### PR DESCRIPTION
**Pull request by RobertBeaudenon**
Issue #11 Development:

- Created two tables one for Loyola and one for SGW.
- Populated it using assets with a predefined schema that matches the entities defined.
- Created related repository and view model classes in order to fetch the data from the DB.

### CODE REVIEW

File: app/src/main/java/com/droidhats/campuscompass/repositories/NavigationRepository.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package com.droidhats.campuscompass.repositories
```
Comment (NicholasNagy) [inline app/src/main/java/com/droidhats/campuscompass/repositories/NavigationRepository.kt]: It's cool that you took the initiative to start off this repository, however, the functionality in this file will be refactored out soon, as this class will become an abstract class for the indoor and outdoor navigation

File: app/src/main/java/com/droidhats/campuscompass/roomdb/AppDB.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+@Database(entities = [ShuttleBus_Loyola_Entity::class,ShuttleBus_SGW_Entity::class],version = 3, exportSchema = false)
+abstract class AppDB : RoomDatabase() {
+
+   //When we invoke this function it will call the DAO class
+    abstract fun shuttleBusDAO(): ShuttleBus_DAO
+
+    companion object {
+        private var INSTANCE: AppDB? = null
+        fun getInstance(context: Context): AppDB? {
+            if (INSTANCE == null) {
```
Comment (NicholasNagy) [inline app/src/main/java/com/droidhats/campuscompass/roomdb/AppDB.kt]: It's nice that we use singleton here, I think it's essential. However, despite the logic being the same, we should use a consistent way for declaring our singletons, unless there is a special reason for that. Please see MapRepository and CalendarRepository for references to other already implemented singletons

File: app/src/main/java/com/droidhats/campuscompass/roomdb/AppDB.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+                synchronized(AppDB::class) {
+                    INSTANCE = Room.databaseBuilder(context,
+                        AppDB::class.java, "CampusCompassDB")
+                        .createFromAsset("database/ShuttleBus.db").fallbackToDestructiveMigration()
+                        .build()
+                }
+            }
+            return INSTANCE
+        }
```
Comment (NicholasNagy) [inline app/src/main/java/com/droidhats/campuscompass/roomdb/AppDB.kt]: we should space out our functions to be consistent

File: app/src/main/java/com/droidhats/campuscompass/roomdb/AppDB.kt [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        fun destroyInstance() {
```
Comment (NicholasNagy) [inline app/src/main/java/com/droidhats/campuscompass/roomdb/AppDB.kt]: I see the use, but we're not using it, and I don't think we have any plan to use a destructor, maybe we should remove this?

File: app/src/main/java/com/droidhats/campuscompass/roomdb/ShuttleBus_SGW_Entity.kt [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import org.jetbrains.annotations.NotNull
+
+@Entity()
+data class ShuttleBus_SGW_Entity(
+
+    @PrimaryKey
+    @NotNull
+    @ColumnInfo(name = "ID")
+    var ID: Int,
+
```
Comment (NicholasNagy) [inline app/src/main/java/com/droidhats/campuscompass/roomdb/ShuttleBus_SGW_Entity.kt]: why double space?

File: app/src/main/java/com/droidhats/campuscompass/roomdb/ShuttleBus_SGW_Entity.kt [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+    @ColumnInfo(name ="SHUTTLE_DAY")
+    @NotNull
+    var shuttle_day: String= "",
+
+
```
Comment (NicholasNagy) [inline app/src/main/java/com/droidhats/campuscompass/roomdb/ShuttleBus_SGW_Entity.kt]: why double space

File: app/src/main/java/com/droidhats/campuscompass/viewmodels/NavigationViewModel.kt [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+import com.droidhats.campuscompass.roomdb.ShuttleBus_Loyola_Entity
+import com.droidhats.campuscompass.roomdb.ShuttleBus_SGW_Entity
+
+class NavigationViewModel: AndroidViewModel{
+
+    private  var navigationRepository: NavigationRepository
+    public  var loyolaShuttleTimes: LiveData<List<ShuttleBus_Loyola_Entity>>
+    private  var sgwShuttleTimes: LiveData<List<ShuttleBus_SGW_Entity>>
+
+    constructor(application: Application) : super(application){
```
Comment (NicholasNagy) [inline app/src/main/java/com/droidhats/campuscompass/viewmodels/NavigationViewModel.kt]: you should leave a space between the super call and the curly bracket

File: app/src/main/java/com/droidhats/campuscompass/viewmodels/NavigationViewModel.kt [hunk 8]
```diff
@@ -0,0 +1,29 @@
+package com.droidhats.campuscompass.viewmodels
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import com.droidhats.campuscompass.repositories.NavigationRepository
+import com.droidhats.campuscompass.roomdb.ShuttleBus_Loyola_Entity
+import com.droidhats.campuscompass.roomdb.ShuttleBus_SGW_Entity
+
+class NavigationViewModel: AndroidViewModel{
```
Comment (NicholasNagy) [inline app/src/main/java/com/droidhats/campuscompass/viewmodels/NavigationViewModel.kt]: you should leave a space between AndroidViewModel and the open bracket

File: app/src/main/java/com/droidhats/campuscompass/roomdb/ShuttleBus_Loyola_Entity.kt [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    @PrimaryKey
+    @NotNull
+    @ColumnInfo(name = "ID")
+    var ID: Int,
+
+    @NotNull
+    @ColumnInfo(name ="SHUTTLE_DAY")
+    var shuttle_day: String= "",
+
+
```
Comment (NicholasNagy) [inline app/src/main/java/com/droidhats/campuscompass/roomdb/ShuttleBus_Loyola_Entity.kt]: why double space?

File: app/src/main/java/com/droidhats/campuscompass/roomdb/ShuttleBus_SGW_Entity.kt [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+    @ColumnInfo(name ="SHUTTLE_TIME")
+    @NotNull
+    var shuttle_time: String= ""
+
+) {
+
+
```
Comment (NicholasNagy) [inline app/src/main/java/com/droidhats/campuscompass/roomdb/ShuttleBus_SGW_Entity.kt]: why have all this spacing?

File: app/src/main/java/com/droidhats/campuscompass/roomdb/ShuttleBus_SGW_Entity.kt [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+    @ColumnInfo(name ="SHUTTLE_DAY")
+    @NotNull
+    var shuttle_day: String= "",
+
+
+    @ColumnInfo(name ="SHUTTLE_TIME")
+    @NotNull
+    var shuttle_time: String= ""
+
```
Comment (NicholasNagy) [inline app/src/main/java/com/droidhats/campuscompass/roomdb/ShuttleBus_SGW_Entity.kt]: why leave this space here?

File: app/src/main/java/com/droidhats/campuscompass/roomdb/ShuttleBus_Loyola_Entity.kt [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+    @NotNull
+    @ColumnInfo(name ="SHUTTLE_TIME")
+    var shuttle_time: String= ""
+
+){}
+
```
Comment (NicholasNagy) [inline app/src/main/java/com/droidhats/campuscompass/roomdb/ShuttleBus_Loyola_Entity.kt]: why 2 spaces at the end of the file?

File: app/src/main/java/com/droidhats/campuscompass/roomdb/AppDB.kt [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+            INSTANCE = null
+        }
+    }
+
+
```
Comment (NicholasNagy) [inline app/src/main/java/com/droidhats/campuscompass/roomdb/AppDB.kt]: why double space?

File: app/src/main/java/com/droidhats/campuscompass/roomdb/AppDB.kt (hunk 2 above)
Comment (RobertBeaudenon) [reply app/src/main/java/com/droidhats/campuscompass/roomdb/AppDB.kt]: Fixed.

File: app/src/main/java/com/droidhats/campuscompass/roomdb/AppDB.kt (hunk 3 above)
Comment (RobertBeaudenon) [reply app/src/main/java/com/droidhats/campuscompass/roomdb/AppDB.kt]: Fixed due to previous change.

File: app/src/main/java/com/droidhats/campuscompass/roomdb/AppDB.kt (hunk 4 above)
Comment (RobertBeaudenon) [reply app/src/main/java/com/droidhats/campuscompass/roomdb/AppDB.kt]: I removed it.

File: app/src/main/java/com/droidhats/campuscompass/roomdb/AppDB.kt [hunk 14]
```diff
@@ -0,0 +1,30 @@
+package com.droidhats.campuscompass.roomdb
+
+import android.content.Context
+import androidx.room.Database
+import androidx.room.Room
+import androidx.room.RoomDatabase
+import androidx.sqlite.db.SupportSQLiteDatabase
+
+@Database(entities = [ShuttleBus_Loyola_Entity::class,ShuttleBus_SGW_Entity::class],version = 3, exportSchema = false)
+abstract class AppDB : RoomDatabase() {
```
Comment (makchamp) [inline app/src/main/java/com/droidhats/campuscompass/roomdb/AppDB.kt]: Would be preferable to change the name of this file as we will have more than 1 database schema

Comment (RobertBeaudenon) [reply app/src/main/java/com/droidhats/campuscompass/roomdb/AppDB.kt]: Name updated to ShuttleBusDB

**Pull request merged**